lib/world.rb

```diff
@@ -482,6 +482,8 @@ def transform_value(x)
         table( x.raw.map { |row| row.map { |cell| transform_value(cell) } } )
       elsif x.respond_to? :gsub
         x.gsub(/<%=(.+?)%>/m) { |c| eval $1 }
+      elsif x.respond_to? :to_s
+        x.to_s
```

Comment:
I wonder why we could get anything but a string here. Maybe some default processing by Cucumber 5? And also it doesn't happen always as it seems :/

I wonder if we can disable such cucumber automatic stuff.

Reply:
@akostadinov it comes down to the step definition.  If the calling step parameter is Integer, that's when we get into trouble.  Example failure that this PR fixes http://ci-qe-openshift.usersys.redhat.com/userContent/cucushift/v3/2021/05/05/04:45:50/PV_PVC_bind_in_a_reasonable_time_when_PVC_is_created_before_PV_while_PVC_pre-bind_to_PV/console.html

Reply:
one of the failures this PR won't fix is something like below, which deals with nesting RegExp
```    Given I store all machines in the "openshift-machine-api" project to the :machines clipboard # features/step_definitions/resources.rb:209
      Your block takes 5 arguments, but the Regexp matched 4 arguments. (Cucumber::Glue::ArityMismatchError)
      features/step_definitions/resources.rb:209:in `/^(I|admin) stores? all (\w+)( in the(?: "(.+?)")? project)? to the(?: :(\S+))? clipboard$/'```

Reply:
@akostadinov for the above failure, perhaps the step itself needs to be re-written.

Reply:
@akostadinov the regexp checks out with rubular.com https://rubular.com/r/KYwSHJjM7hRiYI  so I'm not sure why cuke5 is complaining

Reply:
There is another failure mode with

https://github.com/openshift/cucushift/blob/master/web/admin-console/home.feature#L60

```
Given /^I repeat the following steps for each #{SYM} in cb\.([\w]+):$/ do |varname, cbsym, steps_str|
  eval_regex = /\#\{(.+?)\}/
  eval_found = steps_str =~ eval_regex
  begin
    logger.dedup_start
    cb[cbsym].each { |x|
      cb[varname] = x
      if eval_found
        steps steps_str.gsub(eval_regex) { |s| "<%= #{$1} %>"}
```

These meta-steps won't work.  `steps_str` is transformed before we get to to `steps_str.gsub`

Reply:
I guess it's bad syntax.  Should be  `#{}` instead of `<%= %>` in the meta-step?

```ruby
    When I repeat the following steps for each :resource in cb.resources:
    """
    When I perform the :goto_resource_page_under_default_project web action with:
      | resource_url_name   | #{cb.resource.downcase.to_s.delete(" ")+"s"} |
    Then the step should succeed
```

Maybe fixed with https://github.com/openshift/cucushift/pull/8530

Reply:
Proper syntax. This is to evaluate the expression by each iteration.

Reply:
wrt `Your block takes 5 arguments, but the Regexp matched 4 arguments.`, please tell the exact scenario failing. It is very strange why it happens in any case.

Reply:
@rbbratta , you are correct,  I read your message improperly.

Reply:
@akostadinov this is the step that's failing ```Given I store all machines in the "openshift-machine-api" project to the :machines clipboard```

Reply:
@akostadinov Rubular ouptut  https://rubular.com/r/KYwSHJjM7hRiYI

Reply:
@pruan so any call to the step or only inside a loop step? Do you have exact scenario?

Reply:
@pruan-rht , ok, it seems like when regular expression matches something from the cucumber's built-in parameter types, then things tend to break. In this situation step definition contained expression `\w+`. So I modified step definition like:
```
Given /^(I|admin) stores? all (#{WORD})( in the#{OPT_QUOTED} project)? to the#{OPT_SYM} clipboard$/
```
And now it seems to catch it. Please try and push such a change. I'll be gone for the rest of the day but will check in the evening for further issue.

i.e. replaced `\w+` with `#{WORD}`